runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -685,9 +713,96 @@ private void removeCounter()
 		counter = null;
 	}
 
+	private void taskLookup(SetMessage setMessage, String message)
+	{
+		if (!config.taskCommand())
+		{
+			return;
+		}
+
+		ChatMessageType type = setMessage.getType();
+
+		final String player;
+		if (type.equals(ChatMessageType.PRIVATE_MESSAGE_SENT))
+		{
+			player = client.getLocalPlayer().getName();
+		}
+		else
+		{
+			player = sanitize(setMessage.getName());
+		}
+
+		String task;
+		try
+		{
+			task = chatClient.getTask(player);
+			Matcher taskMatch = TASK_VALIDATION.matcher(task);
+			if (!taskMatch.matches())
+			{
+				return;
+			}
+		}
+		catch (IOException ex)
+		{
+			log.debug("unable to lookup slayer task", ex);
+			return;
+		}
+
+		String response = new ChatMessageBuilder()
+			.append(ChatColorType.NORMAL)
+			.append("Slayer Task: ")
+			.append(ChatColorType.HIGHLIGHT)
+			.append(task)
+			.build();
+
+		final MessageNode messageNode = setMessage.getMessageNode();
+		messageNode.setRuneLiteFormatMessage(response);
+		chatMessageManager.update(messageNode);
+		client.refreshChat();
+	}
+
+	private boolean taskSubmit(ChatInput chatInput, String value)
+	{
+		if (Strings.isNullOrEmpty(taskName))
+		{
+			return true;
+		}
+		final String playerName = client.getLocalPlayer().getName();
+		// taskName (taskLocation) amount/initialAmount
+		final String task = capsString(taskName) +
+			(Strings.isNullOrEmpty(taskLocation)
+				? ""
+				: " (" + taskLocation + ")") +
+			" " + amount + "/" + initialAmount;
+
+		executor.execute(() ->
+		{
+			try
+			{
+				chatClient.submitTask(playerName, task);
+			}
+			catch (Exception ex)
+			{
+				log.warn("unable to submit slayer task", ex);
+			}
+			finally
+			{
+				chatInput.resume();
+			}
+		});
+
+		return true;
+	}
+
 	//Utils
 	private String capsString(String str)
 	{
 		return str.substring(0, 1).toUpperCase() + str.substring(1);
 	}
+
+	private static String sanitize(String str)
```

Comment:
Deathbeam had this moved within java\net\runelite\client\util\Text.java. I think that would be the best practice to allow for future chat commands to utilize it.